pkg/storage/ingest/reader.go

```diff
@@ -219,6 +219,14 @@ func (r *PartitionReader) processNextFetchesUntilMaxLagHonored(ctx context.Conte
 	})
 
 	for boff.Ongoing() {
+		// Send a direct request to the Kafka backend to fetch the partition start offset.
+		partitionStartOffset, err := r.offsetReader.FetchPartitionStartOffset(ctx)
```

Comment:
Note to reviewers: in theory we don't need to do this inside the `for` loop. I moved it here because it's easier to have retries given by `boff`. It shouldn't make a practical difference, given we expect `FetchPartitionStartOffset()` to run quickly.